processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -308,13 +315,17 @@ public int getValueCardinality()
                           @Override
                           public String lookupName(int id)
                           {
-                            final String retVal = column.lookupName(id);
-                            return retVal == null ? "" : retVal;
+                            final String value = column.lookupName(id);
+                            final String retVal = extractionFn == null ? value : extractionFn.apply(value);
+                            return Strings.nullToEmpty(retVal);
```

Comment:
Ah I might have inadvertantly changed the behavior here, but I will leave the nulls to empty before applying the extraction, since that's what it did before.